### ISSUE TITLE
templates also contain hardcoded recipe_dir

### ISSUE THREAD

**Issue by timsnyder** [conda-forge/conda-smithy#1257]
Issue:

After fixing #1254, further testing revealed that there are hardcodes of 'recipe' instead of `{{ recipe_dir }}` in conda_smithy/templates/*

Failure can be observed in the mangle_compiler step in an osx build:
[![image](https://user-images.githubusercontent.com/83846/76565064-ce1b4180-6477-11ea-9651-105cd2987b89.png)](https://dev.azure.com/sparcians/map/_build/results?buildId=39&view=logs&j=ce7cbaa7-582c-5570-40f4-267b3d85fc9f&t=60b22237-ee6f-5809-55fc-3b629141cc1c)

Unfortunately @scopatz and I got #1254 merged before I realized that it has issues and I'm not sure whether reopening the PR is a good idea.  Creating a new ticket.  

Environment and conda info are same as #1254

**Comment by scopatz** [comment]
Yeah, another PR would be helpful. Thanks @timsnyder!

**Comment by timsnyder** [comment]
You're welcome @scopatz !  One of the hardcodes is in [the azure_pipelines_win.yml.tmpl](https://github.com/conda-forge/conda-smithy/blob/61b54f4c530bb47eba209f1aeb8d49759bad55a2/conda_smithy/templates/azure-pipelines-win.yml.tmpl#L96) .  What I'm working on doesn't build on windows and we've already seen that the unit-tests in conda-smithy don't catch everything.

Should I try manually rerendering one of the feedstocks on a branch to make sure I didn't break windows?  Any suggestions for a quick-building feedstock that gets built on everything?

**Comment by scopatz** [comment]
I think the make feedstock builds relatively fast on all platforms

**Comment by timsnyder** [comment]
Bueno.  I'll make a 'DON'T MERGE ME, JUST TESTIN SMITHY' PR over there with my changes used to rerender before I push here.

**Comment by timsnyder** [comment]
Interesting.  I'm getting errors when I try to rerender the make-feedstock before making any changes and I've reverted my conda-smithy and conda-forge-pinning back to the last MNT: commit's recorded values (conda-build version was already there).

**Comment by scopatz** [comment]
hmmm weird

**Comment by timsnyder** [comment]
I'm going to start an empty PR and see if the admin bot can rerender it.

**Comment by timsnyder** [comment]
So, [the bot was able to rerender it and used newer conda-build](https://github.com/conda-forge/make-feedstock/pull/21#issuecomment-598432887).  I updated conda-build, conda-smithy and cf-pinnings to the latest and still get errors when I rerender locally.

If I fireup a condaforge/linux-anvil-comp7 docker container locally and install everything into the /opt/conda, it renders.

Now, I'm installing a clean osx miniconda and trying with that.  I really hope this isn't some osx-specific thing and that I've just managed to hose my conda installation somehow.
